### ISSUE TITLE
GG 1.3.1 -- HTML Generator loops on a particular source file

### ISSUE THREAD

**Issue by jandac-2020** [DistributedProofreaders/guiguts#859]
Working on a project 'Historical Record of the Thirty-first ...'. After doing the Split, and completing the .txt version, it was time to do the HTML. I load up hr31hunt-split.htm and fire up 'Autogenerate HTML'. For the first time ever (for me), GG does not complete its generation but loops in the section "Marking footnote blocks".

Probably some error in my file, but I do not see  what the issue is.  But GG should not loop even on a bad file.

[hr31.zip](https://github.com/DistributedProofreaders/guiguts/files/7551360/hr31.zip)

**Comment by jandac-2020** [comment]
Update.
Two Footnotes (6 and 37) ended with
`xxx
*/
]`
I inserted a blank line after the */ and tried Autogenerate again. It still looped in the same way.
I then removed those two Footnotes, and changed [6] and [37] to {6} and {37}.
I tried Autogenerate again -- and it works :-)
So I can handcraft those two Footnotes back into my *.htm file and move forward.

Still curious as to why GG loops in the first place.

cheers
JohnC

**Comment by windymilla** [comment]
I think I've found the error. If you want to try the fix, you could edit HTMLConvert.pm, if you feel confident doing that. There is a line (line 1154 if you are using version 1.3.2, or similar if you aren't). It currently says:
```my $pdiv = $textwindow->get( $endoflastfootnoteinblock . '-6c',```
and you should change the 6 to an 8, i.e. 
```my $pdiv = $textwindow->get( $endoflastfootnoteinblock . '-8c',```
Note that the following line has a ```+6c``` in it, but that should remain a 6.

**Comment by windymilla** [comment]
Ironically, it was an edit I made in response to your own bug report [here](https://www.pgdp.net/phpBB3/viewtopic.php?p=1233972#p1233961) that wasn't quite good enough. I introduced the idea of searching back 6 characters to make sure that a footnote closing div from the bottom of a page was placed before the next page marker. However, if that footnote contained something like `/*` markup at the end, there could be an extra blank line which meant I really needed to search back 7 characters. I've set it to 8 to be on the safe side :)
